src/algorithms/calorimetry/CalorimeterHitReco.cc

```diff
@@ -44,6 +44,14 @@ void CalorimeterHitReco::init(const dd4hep::Detector* detector, const dd4hep::re
     m_log = logger;
 
     // threshold for firing
+    // Should set either m_cfg.thresholdFactor or m_cfg.thresholdValue, not both
+    if ( m_cfg.thresholdFactor > 0 &&  m_cfg.thresholdValue > 0){
```

Comment:
Technically, these could be specified as negative (no unsigned float after IEEE 754).
```suggestion
    if ( m_cfg.thresholdFactor != 0 &&  m_cfg.thresholdValue != 0){
```

Reply:
Another approach here commonly taken is:
```suggestion
    if ( m_cfg.thresholdFactor * m_cfg.thresholdValue != 0){
```
but that may be slightly worse performance-wise (like we care in an init method... but this is deeply trained into me, so I have to mention it).

Reply:
I'm actually hypersensitive to float `==` or `!=` comparisons, although probably not warranted for 0. I usually resort to something like `fabs(a-b)>1e-9`. Anyway, whichever you prefer is fine.

Reply:
Yes, excellent instincts, but indeed not warranted for comparisons with default initialized values that can be represented exactly and cheaply in floating point.

Reply:
If you want to use 1e-9, then at least use the epsilon from numerical_limits.

Reply:
Just an example, but yes, I've gone the extra step before to use numerical_limits. Not worth it here, 1e-6 should be fine (it's late, I confused 6 and 9 :) ) .